log2ram

```diff
@@ -70,25 +70,30 @@ case "$1" in
       mount --bind $RAM_LOG/ $HDD_LOG/
       mount --make-private $HDD_LOG/
       wait_for $HDD_LOG
-	if [ "$ZL2R" = true ]; then
+	  if [ "$ZL2R" = true ]; then
 		createZramLogDrive
-		mount -t ext4 -o nosuid,noexec,nodev,user=log2ram /dev/zram${RAM_DEV} ${RAM_LOG}/
-	else
+		mount -t ext4 -o nosuid,noexec,nodev,discard,user=log2ram /dev/zram${RAM_DEV} ${RAM_LOG}/
+	  else
 		mount -t tmpfs -o nosuid,noexec,nodev,mode=0755,size=${SIZE} log2ram $RAM_LOG/
-	fi
+	  fi
       wait_for $RAM_LOG
       syncFromDisk
       ;;
 
   stop)
       syncToDisk
-	#ZRAM_LOG=$(awk '$2 == "/var/log" {print $1}' /proc/mounts)
-	#ZRAM_LOG=$(echo ${ZRAM_LOG} | grep -o -E '[0-9]+')
       umount -l $RAM_LOG/
       umount -l $HDD_LOG/
-	# Unsure as even with Root permision denied
-	#echo ${ZRAM_LOG} > /sys/class/zram-control/hot_remove
-      ;;
+	  if [ "$ZL2R" = true ]; then
+		RAM_DEV=$(cat /sys/class/zram-control/hot_add)
+		if [ "$RAM_DEV" -eq "1" ];then
+			rmod zram
+		else
+			RAM_DEV=$(df /var/log | tail -1 | awk '{ print $1 }' | tr -dc '0-9')
+			echo "$RAM_DEV" > /sys/class/zram-control/hot_remove
```

Comment:
If we reboot, is it useful to remove the zram ?

Reply:
Its clean